packages/itwinui-react/src/core/Tabs/Tabs.tsx

```diff
@@ -64,547 +43,759 @@ type TabsOrientationProps =
 
-      {actions && (
-        <Box className='iui-tabs-actions-wrapper'>
-          <Box className='iui-tabs-actions'>{actions}</Box>
-        </Box>
-      )}
+// ----------------------------------------------------------------------------
+// Tabs.Panel component
+
+type TabsPanelOwnProps = {
+  /**
+   * Value used to associate the panel with a given tab.
+   */
+  value: string;
+  /**
+   * @deprecated Don't pass `id`, as it will be automatically set.
+   */
+  id?: string;
+};
+
+const TabsPanel = React.forwardRef((props, ref) => {
+  const { value, className, children, ...rest } = props;
+
+  const { activeValue, idPrefix } = useSafeContext(TabsContext);
+
+  return (
+    <Box
+      className={cx('iui-tabs-content', className)}
+      aria-labelledby={`${idPrefix}-tab-${value}`}
+      role='tabpanel'
+      hidden={activeValue !== value ? true : undefined}
+      id={`${idPrefix}-panel-${value}`}
+      ref={ref}
+      {...rest}
+    >
+      {children}
+    </Box>
+  );
+}) as PolymorphicForwardRefComponent<'div', TabsPanelOwnProps>;
+TabsPanel.displayName = 'Tabs.Panel';
+
+// ----------------------------------------------------------------------------
+// Tabs legacy component
+
+type TabsLegacyProps = {
+  /**
+   * Content displayed to the right/bottom of the horizontal/vertical tabs
+   *
+   * If `type = 'pill'`, `actions` is not applicable.
+   */
+  actions?: React.ReactNode[];
+  /**
+   * Elements shown for each tab.
+   * Recommended to pass an array of `Tab` components.
+   */
+  labels: React.ReactNode[];
+  /**
+   * Handler for activating a tab.
+   */
+  onTabSelected?: (index: number) => void;
+  /**
+   * Index of the active tab.
+   */
+  activeIndex?: number;
+  /**
+   * Control whether focusing tabs (using arrow keys) should automatically select them.
+   * Use 'manual' if tab panel content is not preloaded.
+   * @default 'auto'
+   */
+  focusActivationMode?: 'auto' | 'manual';
+  /**
+   * Color of the bar on the active tab.
+   * @default 'blue'
+   */
+  color?: 'blue' | 'green';
+  /**
+   * Custom CSS class name for tabs.
+   */
+  tabsClassName?: string;
+  /**
+   * Custom CSS class name for tab panel.
+   */
+  contentClassName?: string;
+  /**
+   * Custom CSS class name for the tabs wrapper.
+   */
+  wrapperClassName?: string;
+  /**
+   * Content inside the tab panel.
+   */
+  children?: React.ReactNode;
+  /**
+   * @deprecated Tabs can be scrollable by default
+   *
+   * Options that can be specified to deal with tabs overflowing the allotted space.
+   */
+  overflowOptions?: {
+    /**
+     * @deprecated
+     *
+     * Whether to allow tabs list to scroll when there is overflow,
+     * i.e. when there is not enough space to fit all the tabs.
+     *
+     * Not applicable to types `pill` and `borderless`.
+     */
+    useOverflow?: boolean;
+  };
+
+  // To remove `defaultValue` and `defaultChecked` from `<div>` props.
+  defaultValue?: never;
+  defaultChecked?: never;
```

Comment:
Same comment as https://github.com/iTwin/iTwinUI/pull/1548#discussion_r1350373178

Reply:
Updated comments for both